.pre-commit-hooks.yaml

```diff
@@ -95,3 +95,11 @@
   language: script
   entry: hooks/yalc/yalc-check.sh
   files: package\.json
+######################
+# Spell checking related hooks
+- id: cspell
+  name: "Check for spelling errors with `cspell-cli`"
+  description: "Find and report spelling errors in staged files"
+  language: script
+  entry: hooks/cspell/cspell.sh
```

Comment:
I'm not sure if there's a way to run the `cspell` pre-commit hook like how cspell suggests  https://github.com/streetsidesoftware/cspell-cli/blob/main/.pre-commit-hooks.yaml (instead of running an executable bash script)

My theory is the way above would allow us to store the dictionary or cspell.json in this repo.

In the current implementation I believe it may be difficult to create a shared dictionary. [See concerns / comment #2 in `cspell.sh`](https://github.com/turo/pre-commit-hooks/pull/42/files#r1451062620)

Reply:
I think you are correct: I looked into making a `cspell` id for repos to reference from, but I think we'd largely have to fork/duplicate the setup in https://github.com/streetsidesoftware/cspell-cli in this repo, which seemed like a poor trade-off for avoiding the following lines of logic in the repos:

```yaml
repos:
  - repo: https://github.com/streetsidesoftware/cspell-cli
    rev: v8.3.0
    hooks:
      - id: cspell
```

The only benefit I could see, beyond convenience, is maintaining a single version of `cspell` behind on our versioned releases. The overhead for that benefit seems dubious. 🤷